yara-x/src/modules/string.rs

```diff
@@ -0,0 +1,86 @@
+use crate::modules::prelude::*;
+use crate::modules::protos::string::*;
+
+#[module_main]
+fn main(_ctx: &ScanContext) -> String {
+    // Nothing to do, but we have to return our protobuf
+    let string_proto = String::new();
+    string_proto
+}
+
+#[module_export(name = "to_int")]
+fn to_int(ctx: &ScanContext, string: RuntimeString) -> Option<i64> {
+    let string = string.as_bstr(ctx);
+
+    match string.to_str() {
+        Ok(n) => return Some(n.parse::<i64>().unwrap()),
+        Err(_) => return None,
+    }
+}
+
+#[module_export(name = "to_int")]
+fn to_int_base(
+    ctx: &ScanContext,
+    string: RuntimeString,
+    base: i64,
+) -> Option<i64> {
+    let string = string.as_bstr(ctx);
+
+    if base < 2 || base > 36 {
+        return None;
+    }
+    let strobj = match string.to_str() {
```

Comment:
Similarly, this could be simplified by using `?` and `.ok()`

```rust
fn to_int_base(
    ctx: &ScanContext,
    string: RuntimeString,
    base: i64,
) -> Option<i64> {
    if base < 2 || base > 36 {
        return None;
    }
    let string = string.as_bstr(ctx).to_str().ok()?;
    i64::from_str_radix(string, base as u32).ok()
}
```

Notice that you don't need to use `.try_into().unwrap()`, firstly because you already made sure that `base64` is between 2 and 36, so it's safe to convert it to an `u32`, and secondly because `base.try_into().unwrap()` is not safer than `base as u32`, both will panic if the conversion is not possible. `.try_into()` makes sense when you handle the error.

Reply:
Thanks. I will get it fixed.

Reply:
It's not true for the `as`, this is a truncation as would be done in C. So a safe way to cast would be with a try_into.

```
let base: u32 = base.try_into().ok()?;
if base < 2 || base > 36 {
    return None;
}
```

Reply:
Fixed to @plusvic suggestion. Any objections to this @plusvic ? Else i can change it to that

Reply:
Updated code to 
```
    let base: u32 = base.try_into().ok()?;
    if !(2..=36).contains(&base) {
```